vllm/entrypoints/openai/tool_parsers/mistral_tool_parser.py

```diff
@@ -185,185 +203,357 @@ def extract_tool_calls_streaming(
         request: ChatCompletionRequest,
     ) -> Union[DeltaMessage, None]:
 
-        # if the tool call token is not in the tokens generated so far, append
-        # output to contents since it's not a tool
         if self.bot_token not in current_text:
+            # if the tool call token is not in the tokens generated so far,
+            # append output to contents since it's not a tool
             return DeltaMessage(content=delta_text)
 
         # if the tool call token ID IS in the tokens generated so far, that
         # means we're parsing as tool calls now
+        if _is_fn_name_regex_support(self.model_tokenizer):
```

Comment:
Naming of this function could be modified to be more readable as the streaming function does not use this .

Basically this is used here to know if the tool calls should be parsed with :
- the old parser (tool call like `[TOOL_CALLS][{"name":"add" , "arguments":{"a": 3, "b": 4} } ]`)
- the new parser (tool call like `[TOOL_CALLS]add{"a": "3", "b": "4"}`)